src/abstractdataframe/sort.jl

```diff
@@ -336,13 +336,40 @@ Sort.defalg(df::AbstractDataFrame, o::Ordering; alg=nothing, cols=[]) =
              lt=isless, by=identity, rev::Bool=false, order::Ordering=Forward)
 
 Test whether data frame `df` sorted by column(s) `cols`.
+If no column is specified, it checks if the `df` is sorted lexicographically.
```

Comment:
This is an important comment. Can you please add it also to `sort` and `sort!`. Also can you make it more precise that "no column is specified" means "column selector selects no columns".

@nalimilan - this rule is non-intuitive, but changing it would be breaking. Should we deprecate this behavior for 2.0 release (it would be a separate PR).

Reply:
@Chandu-4444 - if what I write is not clear please comment and I will give you examples.

Reply:
> Can you please elaborate?

`sort` and related functions were developed in very early days of DataFrames.jl when there was no precise rules for the whole ecosystem. Therefore some inconsistencies slipped through.

A proper definition of `sort` should be (I am skipping kwargs):
```
Base.sort(df::AbstractDataFrame, cols=All())
```
however it is
```
Base.sort(df::AbstractDataFrame, cols=[])
```
and as a consequence if you pass any column selector that selects no columns the data frame is sorted lexicographically.

While technically it should happen if you select `All()` columns. If you select no columns a proper behavior should be that the data frame is not reordered (it is left as is as you have not requested on what the data frame should be sorted).

We have two options:
* leave things as is forever (probably there is not much harm with that as it is unlikely that someone does sorting and wants the data frame to be left untouched)
* make a deprecation of how selecting no columns is handled and in 2.0 release change it the way I describe above.

Reply:
I get it now, thanks for that excellent explanation. Perhaps this column selector can be made a required argument (Like the one Python has in Pandas)?

Reply:
> Perhaps this column selector can be made a required argument

This would be even more breaking. By proposing the change to:
```
Base.sort(df::AbstractDataFrame, cols=All())
```
we are not breaking calls like `sort(df)` which are quite likely to exist in current user code.

We try to limit breaking changes. Calls like `sort(df, [])` are quite unlikely, so we could deprecate them and change their behavior in 2.0.

Let us wait for an opinion from @nalimilan.

Reply:
That's indeed unfortunate. I'm not sure we should make a breaking change though given that it's not super likely to confuse people (if you call `sort` with no columns, it's most likely a mistake anyway and you'll get unexpected results), so I'd opt for a deprecation.

Reply:
OK - let us then document that passing no columns is an undefined behavior.